src/table/cells/handle-key-press.js

```diff
@@ -102,12 +98,28 @@ export const bodyHandleKeyPress = (evt, rootElement, rowIndex, colIndex, cell, s
     case 'Enter': {
       preventDefaultBehavior(evt);
       selState.api.confirm();
+      // TODO: make sure focus is kept, or set on nebula cell
       break;
     }
-    // Esc: Cancels selections.
+    // Esc: Cancels selections if any, otherwise blur
     case 'Escape': {
       preventDefaultBehavior(evt);
-      selState.api.cancel();
+      if (selState?.rows.length) {
+        selState.api.cancel();
+        // TODO: make sure focus is kept
```

Comment:
We shouldn't have to do anything more here really. AFAIK if you have proper keys sets for all sibling elements, react should preserve the state of the element, including focus, when updating. I'll look further into this in a different PR